### DISCUSSION TITLE
Adding a device with curl using a specific namespace and/or tag

### DISCUSSION THREAD

**Discussion by ghost** [shellhub-io/shellhub#3099] [Ideas]
Dear Shellhub Team,

You have built an incredible product. Due to remote port forwarding and vscode shellhub is becoming the choice for remote ssh device management. In order to use shellhub in production environments, I realized that is super easy to register new devices via the

curl -sSf "https://cloud.shellhub.io/install.sh?tenant_id=XXX" | sh

command, however to use it in automatic deployment scripts it would be great to add two additional parameters to your
REST API, i.e. namespace and tag.

curl -sSf "https://cloud.shellhub.io/install.sh?tenant_id=YOUR_ID&tag=YOUR_TAG&namespace=YOUR_NAMESPACE" | sh

Using this command the device will be added to the given namespace and/or tag, helping admins to easily identify the customers network devices and purpose. This is important for industrial usage, for example if more than 10 devices are registered at the same time by deployment scripts, it will help you to differentiate those devices in the admin web panel. Of course care must be taken to make sure that the namespace is created or does not exist or an default tag/namespace is added, when not added to the REST call.

It would be great to add such a feature, in order to register devices automatically for specific customers.

Thanks in advance

Flapjack Team

**Comment by gustavosbarreto** [comment]
@fj-mzoehr

We greatly appreciate the idea you've shared with us.

To better understand your use case and how we can tailor this feature to suit your needs effectively, could you kindly provide us with more details?

Additionally, we want to inform you that it is indeed possible to add tags to a device via our API. You can find comprehensive information on how to achieve this in our API documentation, which can be accessed at https://docs.shellhub.io/developers/openapi/community/.
